### PR TITLE
Add tests for plugin resolution within a monorepo setup

### PR DESCRIPTION
**Pull request by rwjblue**
Adds tests inspired by those added in #2359, but without the need to `git clone` + `yarn install` + `yarn build` external packages in the test harness.

### CODE REVIEW

**Comment by rwjblue** [comment]
These tests pass without changing anything about the implementation.

@NullVoxPopuli - Can you copy one of these tests and tweak it in order to fail? I'm still not sure what is actually broken in your use case...

File: lib/get-config.js [hunk 1]
```diff
@@ -39,7 +40,9 @@ async function requirePlugin(workingDir, pluginName, fromConfigPath) {
   let pluginPath = resolve.sync(pluginName, { basedir, extensions: ['.js', '.mjs', '.cjs'] });
 
   try {
-    const { default: plugin } = await import(pluginPath);
+    let pluginURL = pathToFileURL(pluginPath);
```
Comment (rwjblue) [inline lib/get-config.js:43]: Note: this definitely fixes a bug in plugin resolution, but only for windows users (we previously attempted to `import('c:/foo/bar/baz.js')` which is not allowed, you have to do `import('file:///c:/foo/bar/baz.js')`).

File: test/acceptance/monorepo-test.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+import run from '../helpers/run.js';
+import setupEnvVar from '../helpers/setup-env-var.js';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+describe('monorepo setups', function () {
+  setupEnvVar('FORCE_COLOR', '0');
+  setupEnvVar('LC_ALL', 'en_US');
+
+  // Fake project
```
Comment (scalvert) [inline test/acceptance/monorepo-test.js:16]:
```suggestion
```

**Review by scalvert — APPROVED** [review]

File: lib/get-config.js (hunk 1 above)
Comment (bmish) [reply lib/get-config.js:43]: This should be labeled as a `bug fix` then right?

**Pull request merged**